### DISCUSSION TITLE
How Can I use template string to import image assets dynamic?

### DISCUSSION THREAD

**Discussion by Alkaidcc** [web-infra-dev/rsbuild#2092] [Q&A]
```vue
<template>
	<img :src="getImageUrl(props.icon)" alt="">
</template>

<script>
export default {
	props: ['icon']
	methods: {
		getImageUrl(icon){
			// How can I return the source URL to dev and pord? It's not working.
			return `@/assets/image/${props.icon}.png`
		}
	}
}
</script>
```

**Comment by chenjiahan** [comment]
You can place these assets in the public folder and reference them with URLs.

See: https://rsbuild.dev/guide/basic/static-assets#public-folder

**Reply by Alkaidcc** [reply]
will assets in public compress during bundle? Can I optimize asset size in the public folder?

**Reply by chenjiahan** [reply]
The assets in the public folder will not be bundled or compressed by Rsbuild.

You can compress the image assets in the public folder via https://tinypng.com/

**Reply by Alkaidcc** [reply]
I tried to do that with something like 
```js
const getImgUrl = (name) => {
  return new URL(`../../assets/${name}.png`, import.meta.url).href;
};
```
It has worked well in [Vite](https://cn.vitejs.dev/guide/assets.html#new-url-url-import-meta-url). Does Rsbuild have any plan for a similar feature in the future?

**Reply by chenjiahan** [reply]
This usage is also allowed in Rsbuild, but Webpack/Rspack will replace `import.meta.url` with the absolute `file:` URL of the module. (see https://webpack.js.org/api/module-variables/#importmetaurl), so we can not use the final URL as an img href.

We will conduct research later and confirm whether Rsbuild should support this user
